packages/iota/src/validation/outputs/aliasOutput.ts

```diff
@@ -0,0 +1,156 @@
+import { STATE_CONTROLLER_ADDRESS_UNLOCK_CONDITION_TYPE } from "../../models/unlockConditions/IStateControllerAddressUnlockCondition";
+import { validateFeatures, MAX_METADATA_LENGTH } from "../features/features";
+import { validateNativeTokens } from "../nativeTokens";
+import { IValidationResult, mergeValidationResults } from "../result";
+import { validateUnlockConditions } from "../unlockConditions/unlockConditions";
+import { validateAscendingOrder } from "../validationUtils";
+
+/**
+ * Zero alias id.
+ */
+const ZERO_ALIAS_ID = "0x0000000000000000000000000000000000000000000000000000000000000000";
+/**
+ * Maximum number of features that alias output can have.
+ */
+const MAX_ALIAS_FEATURES_COUNT = 2;
+/**
+ * Maximum number of immutable features that an nft output could have.
+ */
+const MAX_ALIAS_IMMUTABLE_FEATURES_COUNT = 2;
+/**
+ * Maximum number of unlock conditions that alias output can have.
+ */
+const MAX_ALIAS_UNLOCK_CONDITIONS_COUNT = 2;
+
+/**
+ * Validate an alias output.
+ * @param aliasOutput The object to validate.
+ * @param protocolInfo The Protocol Info.
+ * @returns The validation result.
+ */
+export function validateAliasOutput(aliasOutput: IAliasOutput, protocolInfo: INodeInfoProtocol): IValidationResult {
+    const results: IValidationResult[] = [];
+
+    if (aliasOutput.type !== ALIAS_OUTPUT_TYPE) {
+        results.push({
+            isValid: false,
+            errors: [`Type mismatch in alias output ${aliasOutput.type}`]
+        });
+    }
+
+    if (bigInt(aliasOutput.amount).compare(bigInt.zero) !== 1) {
+        results.push({
+            isValid: false,
+            errors: ["Alias output amount field must be larger than zero."]
+        });
+    }
+
+    if (bigInt(aliasOutput.amount).compare(protocolInfo.tokenSupply) === 1) {
+        results.push({
+            isValid: false,
+            errors: ["Alias output amount field must not be larger than max token supply."]
+        });
+    }
+
+    if (aliasOutput.unlockConditions.length !== MAX_ALIAS_UNLOCK_CONDITIONS_COUNT) {
+        results.push({
+            isValid: false,
+            errors: [`Unlock conditions count must be equal to ${MAX_ALIAS_UNLOCK_CONDITIONS_COUNT}.`]
+        });
+    }
+
+    if (aliasOutput.unlockConditions) {
+        if (
+            !aliasOutput.unlockConditions.every(
+                uC =>
+                    uC.type === STATE_CONTROLLER_ADDRESS_UNLOCK_CONDITION_TYPE ||
+                    uC.type === GOVERNOR_ADDRESS_UNLOCK_CONDITION_TYPE
+            )
+        ) {
+            results.push({
+                isValid: false,
+                errors: ["Alias output unlock condition type of an unlock condition must define one of the following types: State Controller Address Unlock Condition and Governor Address Unlock Condition."]
+            });
+        }
+
+        aliasOutput.unlockConditions.map(unlockCondition => {
```

Comment:
`map` usually implies that you return something, so you "map" from one thing to another. I think `forEach` is better here. Or just a `for` loop.